pkg/driver/audiotest/dummy.go

```diff
@@ -15,28 +15,36 @@ import (
 
 func init() {
 	driver.GetManager().Register(
-		&dummy{}, driver.Info{Label: "AudioTest", DeviceType: driver.Microphone},
+		&dummy{state: driver.StateClosed}, driver.Info{Label: "AudioTest", DeviceType: driver.Microphone},
 	)
 }
 
 type dummy struct {
 	closed <-chan struct{}
 	cancel func()
+	state  driver.State
 }
 
 func (d *dummy) Open() error {
-	ctx, cancel := context.WithCancel(context.Background())
-	d.closed = ctx.Done()
-	d.cancel = cancel
-	return nil
+	return d.state.Update(driver.StateOpened, func() error {
+		ctx, cancel := context.WithCancel(context.Background())
+		d.closed = ctx.Done()
+		d.cancel = cancel
+		return nil
+	})
 }
 
 func (d *dummy) Close() error {
-	d.cancel()
-	return nil
+	return d.state.Update(driver.StateClosed, func() error {
```

Comment:
it seems like for each of these closed states now, that if closing fails, the camera is not closed and we are an indeterminate state